test_1.js

```diff
@@ -0,0 +1,687 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const puppeteer = require('puppeteer-core');const findChrome = require('./find_chrome');
+const {rpc} = require('../rpc');
+const debugApp = require('debug')('carlo:app');
+const debugServer = require('debug')('carlo:server');
+const {Color} = require('./color');
+const {HttpRequest} = require('./http_request');
+
+const fs = require('fs');
+const util = require('util');
+const {URL} = require('url');
+const EventEmitter = require('events');
+const fsReadFile = util.promisify(fs.readFile);
+
+let testMode = false;
+
+class App extends EventEmitter {
+  /**
+   * @param {!Puppeteer.Browser} browser Puppeteer browser
+   * @param {!Object} options
+   */
+  constructor(browser, options) {
+    super();
+    this.browser_ = browser;
+    this.options_ = options;    this.windows_ = new Map();    this.exposedFunctions_ = [];
+    this.pendingWindows_ = new Map();
+    this.windowSeq_ = 0;
+    this.www_ = [];
+  }
+
+  async init_() {
+    debugApp('Configuring browser');
+    let page;
+    await Promise.all([
+      this.browser_.target().createCDPSession().then(session => {
+        this.session_ = session;
+        if (this.options_.icon)
```

Comment:
format: style mismatch:
```suggestion
        if (this.options_.icon)this.setIcon(this.options_.icon);
```

⏎␣⁺␣⁺ at column 32 should be removed.